### PR TITLE
Implement 4-digit version scheme 

### PR DESCRIPTION
**Pull request by dokempf**
THe first three digits are the clang-format version, the last digit refers to our wheel packaging. As before, `clang-format_version.cmake` is the single source of truth for these versions.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,11 +17,14 @@ def get_tag(self):
...
-            raise ValueError("Version File not readable")
-        return match.groups()[0]
+        parsed = {}
+        for line in version_file:
+            match = re.match("set\((.*) (.*)\)", line)
+            if len(match.groups()) != 2:
+                raise ValueError("Version File not readable")
+            parsed[match.groups()[0]] = match.groups()[1]
+
+        return f"{parsed['CLANG_FORMAT_VERSION']}.{parsed['CLANG_FORMAT_WHEEL_VERSION']}"
```
Comment (nightlark) [inline setup.py:27]: If the `CLANG_FORMAT_WHEEL_VERSION` is 0 (the first release of the wheel for a particular clang version) maybe the wheel version could be implicit? Are there any side effects for pip installing future versions with wheel-only changes if this were done (@henryiii)?

```suggestion
        if parsed['CLANG_FORMAT_WHEEL_VERSION'] == "0":
            return f"{parsed['CLANG_FORMAT_VERSION']}"
        else:
            return f"{parsed['CLANG_FORMAT_VERSION']}.{parsed['CLANG_FORMAT_WHEEL_VERSION']}"
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -7,6 +7,10 @@ on:
         description: "LLVM version to build"
         required: false
         default: ""
+      wheel_version:
+        description: "Versioning of the wheel packaging (appended to LLVM version)"
```
Comment (nightlark) [inline .github/workflows/release.yml:11]:
```suggestion
        description: "Version of the wheel packaging (appended to LLVM version)"
```

File: setup.py (hunk 1 above)
Comment (henryiii) [reply setup.py:27]: No, there shouldn't be a difference. 1.2.3.0 should be identical to 1.2.3.

Comment (dokempf) [reply setup.py:27]: Good point and thanks to @henryiii for confirming.

**Pull request merged**